pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -129,6 +126,8 @@ func (s *FlinkStateMachine) Handle(ctx context.Context, application *v1alpha1.Fl
 
 	// Update k8s object
 	if updateApplication {
+		now := v1.NewTime(s.clock.Now())
```

Comment:
This change seems unrelated?

Reply:
We are currently not updating LastUpdatedAt when JobStatus/ClusterStatus is modified. 

This seems more logical, to set `LastUpdatedAt` before we make the update call. Happy to open in a different PR/